### PR TITLE
Query: Handle type convert for Round operator

### LINKED ISSUE #7700: InvalidCastException when trying to Round int cast to double
The following query (where `Quantity` is an int property/column)

```C#
ctx.OrderItems.Sum(i => Math.Round((double)i.Quantity, 15)
```
throws
```
info: Microsoft.EntityFrameworkCore.Storage.IRelationalCommandBuilderFactory[1]
      Executed DbCommand (43ms) [Parameters=[], CommandType='Text', CommandTimeout='30']
      SELECT SUM(ROUND([i].[Quantity], 15))
      FROM [OrderItems] AS [i]
fail: Microsoft.EntityFrameworkCore.Query.Internal.QueryCompiler[1]
      An exception occurred in the database while iterating the results of a query.
      System.InvalidCastException: Unable to cast object of type 'System.Int32' to type 'System.Double'.
         at Microsoft.EntityFrameworkCore.Query.QueryMethodProvider.GetResult[TResult](IEnumerable`1 valueBuffers)
         at lambda_method(Closure , QueryContext )
         at Microsoft.EntityFrameworkCore.Query.Internal.QueryCompiler.<>c__DisplayClass20_0`1.<CompileQueryCore>b__0(QueryContext qc)         
System.InvalidCastException: Unable to cast object of type 'System.Int32' to type 'System.Double'.
   at Microsoft.EntityFrameworkCore.Query.QueryMethodProvider.GetResult[TResult](IEnumerable`1 valueBuffers)
   at lambda_method(Closure , QueryContext )
   at Microsoft.EntityFrameworkCore.Query.Internal.QueryCompiler.<>c__DisplayClass20_0`1.<CompileQueryCore>b__0(QueryContext qc)

Unhandled Exception: System.InvalidCastException: Unable to cast object of type 'System.Int32' to type 'System.Double'.
   at Microsoft.EntityFrameworkCore.Query.QueryMethodProvider.GetResult[TResult](IEnumerable`1 valueBuffers)
   at lambda_method(Closure , QueryContext )
   at Microsoft.EntityFrameworkCore.Query.Internal.QueryCompiler.<>c__DisplayClass20_0`1.<CompileQueryCore>b__0(QueryContext qc)
   at EFTest.Program.Main(String[] args)
```

Microsoft.EntityFrameworkCore.SqlServer 1.1.1 

Found when verifying the fix for #7600

### PR DESCRIPTION
**Pull request by smitpatel**
Resolves #7700

### CODE REVIEW

File: src/EFCore.SqlServer/Query/ExpressionTranslators/Internal/SqlServerMathTranslator.cs [hunk 1]
```diff
@@ -84,17 +84,24 @@ public virtual Expression Translate(MethodCallExpression methodCallExpression)
...
                     methodCallExpression.Type,
-                    new[] { methodCallExpression.Arguments[0], Expression.Constant(0), Expression.Constant(1) });
+                    new [] { methodCallExpression.Arguments[0], Expression.Constant(0), Expression.Constant(1) });
             }
 
             if (_roundMethodInfos.Contains(method))
             {
+                var firstArgument = methodCallExpression.Arguments[0];
+
+                if (firstArgument.NodeType == ExpressionType.Convert)
```
Comment (maumar) [inline src/EFCore.SqlServer/Query/ExpressionTranslators/Internal/SqlServerMathTranslator.cs]: test Truncate (and other math methods) also

File: src/EFCore.SqlServer/Query/ExpressionTranslators/Internal/SqlServerMathTranslator.cs [hunk 2]
```diff
@@ -19,51 +19,51 @@ public class SqlServerMathTranslator : IMethodCallTranslator
...
-            { typeof(Math).GetRuntimeMethod(nameof(Math.Sin), new[] { typeof(double) }), "SIN" },
-            { typeof(Math).GetRuntimeMethod(nameof(Math.Tan), new[] { typeof(double) }), "TAN" },
-            { typeof(Math).GetRuntimeMethod(nameof(Math.Sign), new[] { typeof(decimal) }), "SIGN" },
-            { typeof(Math).GetRuntimeMethod(nameof(Math.Sign), new[] { typeof(double) }), "SIGN" },
-            { typeof(Math).GetRuntimeMethod(nameof(Math.Sign), new[] { typeof(float) }), "SIGN" },
-            { typeof(Math).GetRuntimeMethod(nameof(Math.Sign), new[] { typeof(int) }), "SIGN" },
-            { typeof(Math).GetRuntimeMethod(nameof(Math.Sign), new[] { typeof(long) }), "SIGN" },
-            { typeof(Math).GetRuntimeMethod(nameof(Math.Sign), new[] { typeof(sbyte) }), "SIGN" },
-            { typeof(Math).GetRuntimeMethod(nameof(Math.Sign), new[] { typeof(short) }), "SIGN" },
+            { typeof(Math).GetRuntimeMethod(nameof(Math.Abs), new [] { typeof(decimal) }), "ABS" },
```
Comment (anpete) [inline src/EFCore.SqlServer/Query/ExpressionTranslators/Internal/SqlServerMathTranslator.cs]: We use `new[]`

Comment (smitpatel) [inline src/EFCore.SqlServer/Query/ExpressionTranslators/Internal/SqlServerMathTranslator.cs]: I will check rules and update them.

File: src/EFCore.SqlServer/Query/ExpressionTranslators/Internal/SqlServerMathTranslator.cs (hunk 1 above)
Comment (smitpatel) [inline src/EFCore.SqlServer/Query/ExpressionTranslators/Internal/SqlServerMathTranslator.cs]: We are doing it based on case only. Else it is a complex matching problem with multiple elements on both sides.

Comment (maumar) [inline src/EFCore.SqlServer/Query/ExpressionTranslators/Internal/SqlServerMathTranslator.cs]: Truncate is translated to ROUND on SQL, so at least that one should be tested/compensated. I'm fine with skipping the other ones for now, as it indeed would be too much work.

**Comment by smitpatel** [comment]
Updated.

**Pull request merged**